torch/_inductor/codecache.py

```diff
@@ -57,6 +58,36 @@ def cache_dir():
     )
 
 
+class DiskCache:
+    @staticmethod
+    @functools.lru_cache(None)
+    def _subdir():
+        subdir = os.path.join(cache_dir(), "cached_tunings")
+        os.makedirs(subdir, exist_ok=True)
+        return subdir
+
+    @staticmethod
+    @functools.lru_cache(4096)
+    def _read_file(path):
+        with open(path, "r") as fd:
+            return json.loads(fd.read())
+
+    def __init__(self, unique_name):
+        super().__init__()
+        self.unique_name = unique_name
+
+    def lookup(self, key, generate):
+        """
+        Check if we have already generated key, if not call generate()
+        to populate the cache.
+        """
+        path = os.path.join(self._subdir(), code_hash(self.unique_name + repr(key)))
+        if not os.path.exists(path):
+            value = generate()
+            write_atomic(path, json.dumps(value))
+        return self._read_file(path)
```

Comment:
You don't need to read a file and parse JSON if you just wrote it, consider just returning early in the `not os.path_exists` case

Reply:
While this is true, it could create some pretty subtle bugs since the returned value might be slightly different on a cache hit versus a cache miss.  Specifically json will change aliasing relationships.